code/weapon/beam.cpp

```diff
@@ -3044,32 +3047,24 @@ void beam_handle_collisions(beam *b)
 		// add to the recent collision list
 		r_coll[r_coll_count].c_objnum = target;
 		r_coll[r_coll_count].c_sig = Objects[target].signature;
-		r_coll[r_coll_count].c_stamp = -1;
 		r_coll[r_coll_count].cinfo = b->f_collisions[idx].cinfo;
 		r_coll[r_coll_count].quadrant = -1;
 		r_coll[r_coll_count].is_exit_collision = 0;
 		
+		// get the real damage that should be done
+		damage_time_mod = (flFrametime * 1000.0f) / i2fl(BEAM_DAMAGE_TIME);
+		real_damage = real_damage * damage_time_mod;
```

Comment:
This means that the actual damage dealt will decrease with every collision. Is that also what happens in the current code? I cannot see which code in the current version would be responsible for this.